From 0a50becec3a3cca71e575dfec35d69ab295c85cc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?G=C3=B6kt=C3=BCrk=20Y=C3=BCksek?= <gokturk@gentoo.org>
Date: Tue, 3 Apr 2018 23:06:20 -0400
Subject: [PATCH] Platform/Unix: include <sys/sysmacros.h> for major/minor
 macros
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Starting with glibc 2.26, macros "major" and "minor" are only
available from <sys/sysmacros.h> [0]. The build fails with the
following without including this header:

Unix/FilesystemPath.cpp:84:49: error: ‘major’ was not declared in this scope
Unix/FilesystemPath.cpp:84:113: error: ‘minor’ was not declared in this scope

[0] https://sourceware.org/ml/libc-alpha/2017-02/msg00079.html
---
 src/Platform/Unix/FilesystemPath.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Platform/Unix/FilesystemPath.cpp b/src/Platform/Unix/FilesystemPath.cpp
index d6c9cd2..7c28af4 100644
--- a/Platform/Unix/FilesystemPath.cpp
+++ b/Platform/Unix/FilesystemPath.cpp
@@ -15,6 +15,7 @@
 #include "Platform/StringConverter.h"
 #include <stdio.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 
 namespace VeraCrypt
 {
-- 
2.16.3

